extensions/amp-viewer-integration/0.1/test/test-highlight-handler.js

```diff
@@ -409,57 +412,79 @@ describes.realWin.skip(
       expect(setScrollTopStub).to.be.calledOnce;
       expect(setScrollTopStub.firstCall.args[0]).to.equal(350);
     });
+  }
+);
 
+describes.realWin(
+  'HighlightHandler',
+  {
+    // We can not overwrite win.location with realWin.
+    amp: {
+      ampdoc: 'single',
+    },
+  },
+  (env) => {
     // TODO(dmanek): remove `ifChrome` once we remove Chrome version detection
-    it.configure()
+    describe
+      .configure()
       .ifChrome()
-      .run('should highlight using text fragments for Chrome 93', async () => {
-        const {ampdoc} = env;
-        const platform = Services.platformFor(ampdoc.win);
-        env.sandbox.stub(platform, 'isChrome').returns(true);
-        env.sandbox.stub(platform, 'getMajorVersion').returns(93);
-        let whenFirstVisiblePromiseResolve;
-        const whenFirstVisiblePromise = new Promise((resolve) => {
-          whenFirstVisiblePromiseResolve = resolve;
+      .run('with Chrome >= 93', () => {
+        let root = null;
+
+        beforeEach(() => {
+          const {document} = env.win;
+          root = document.createElement('div');
+          document.body.appendChild(root);
+          const div0 = document.createElement('div');
+          div0.textContent = 'text in amp doc';
+          root.appendChild(div0);
+          const div1 = document.createElement('div');
+          div1.textContent = 'highlighted text';
+          root.appendChild(div1);
+
+          //  Used in Chrome 93+
+          env.sandbox.stub(env.ampdoc, 'whenFirstVisible');
+
+          const platform = Services.platformFor(env.ampdoc.win);
+          if (platform.isChrome()) {
```

Comment:
Could potentially clean up the naming of "with Chrome >= 93" by instead having each test stub its version instead of living in the beforeEach

Reply:
Done - renamed the test suite.